onnxruntime/contrib_ops/cuda/math/bias_softmax_impl.cu

```diff
@@ -249,29 +249,31 @@ void DispatchBiasSoftMaxForwardViaDnnLibraryImpl(
   auto* Y_data = reinterpret_cast<CudaT*>(Y->template MutableData<T>());
 
   // binary elementise kernel requires input pitches
-  TArray<int64_t> lhs_padded_strides(X_shape.NumDimensions());
-  for (int i = -1, lhs_pitch = 1; i >= -(int)X_shape.NumDimensions(); i--) {
+  TArray<int64_t> lhs_padded_strides(static_cast<int>(X_shape.NumDimensions()));
+  int64_t lhs_pitch = 1, rhs_pitch = 1;
+  for (int i = -1; i >= -(int)X_shape.NumDimensions(); i--) {
```

Comment:
>int i [](start = 7, length = 5)

Why not the other way around or at least size_t which selects the correct size for the arch?

Reply:
size_t  is unsigned. Here the init value of i is "-1", which can't be fit in size_t. "i" could be ssize_t.

Reply:
And the only change I made here was moving the declaration of "lhs_pitch" out of the loop. I don't quite understand this piece of code.